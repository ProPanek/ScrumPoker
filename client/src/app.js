import React, { Component } from "react";
import {inject, observer} from "mobx-react";
import { withRouter } from "react-router-dom";
import { Grid } from "@material-ui/core";
import Header from "./components/static/header";
import Router from "./components/router/router";
import Notification from "./components/notification/notification"

class App extends Component {
  constructor(props){
    super(props)
    this.props.store.socketStore.initialize()
    this.props.store.userStore.initialize()
    this.props.store.roomStore.initialize()
    this.props.store.jiraStore.initialize()
  }

  render() {
    return (
      <div className="App">
        <Notification />
        <Header />
          <Grid container>
           <Router />
          </Grid>
      </div>
    );
  }
}
export {App};
export default inject("store")(withRouter(observer(App)));

