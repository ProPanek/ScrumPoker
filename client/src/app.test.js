import React from "react";
import {App} from "./app";
import { shallow } from "enzyme";

const noop = () => {}
it("renders without crashing", () => {
    const store = {
        userStore: {initialize: noop},
        roomStore:{initialize: noop},
        socketStore:{initialize: noop},
        jiraStore:{initialize: noop},
    }
    const component = shallow(<App store={store} />)
    expect(component).toMatchSnapshot()
});
