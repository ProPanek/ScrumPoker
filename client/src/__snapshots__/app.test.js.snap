// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <inject-Notification-with-store />
  <inject-withRouter(Header)-with-store />
  <WithStyles(Grid)
    container={true}
  >
    <inject-withRouter(Router)-with-store />
  </WithStyles(Grid)>
</div>
`;
